File: service/src/main/scala/com/mfed/model/GameMap.scala
package com.mfed.model
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

/**
  * Created by Maciej Fedorowiat 
  * on 28/07/2016 02:22.
  * mfedorowiat@gmail.com
  */

@Document
case class GameMap(@Id id: String, levelNumber: Integer, width: Integer, height: Integer, obstacles: java.util.List[(Int, Int)],
                   introductionText: java.util.List[String], functions: java.util.List[String]){
  def produceInitialState() = GameMapState(this, RobotState())
}

case class Obstacle(x: Integer, y: Integer)

File: web-ui/src/scenes/Level/scenes/Board/game_map.js

function GameMap(map) {
    this.width = map.width;
    this.height = map.height;
    this.obstacles = map.obstacles;
    this.diamonds = map.diamonds;
    return this;
}

GameMap.prototype.openField = function (point) {
    for (let i = 0; i < this.obstacles.length; i++) {
        if (this.obstacles[i].x === point.x && this.obstacles[i].y === point.y) {
            return true;
        }
    }
    return false;
};

GameMap.prototype.diamondField = function (point) {
    for (let i = 0; i < this.diamonds.length; i++) {
        if (this.diamonds[i].x === point.x && this.diamonds[i].y === point.y) {
            return true;
        }
    }
    return false;
};

module.exports = GameMap;


